### PR TITLE
refactor(ava/insight): rename some apis, files and folders of insight

### PR DESCRIPTION
**Pull request by chenluli**
### PR includes

- adjust the structure of `insight/algorithms`, put each algorithm under its corresponding insight-type file and move implementations of algorithms to `data`
- adjust the structure of `insight/visualization`, use the similar structures as `insight/narrative`, including the `generator` module and the `strategy` module. (Just add the folders, implementations to be done)
- modify some api names and types
-  `dimension: sting` -> `dimension: { field: sting}`

before:
截屏2023-03-15 下午3 48 27
after:
截屏2023-03-18 上午11 32 44

### CODE REVIEW

File: packages/ava/src/insight/algorithms/category-outlier/IQR.ts
Comment (pddpd) [inline packages/ava/src/insight/algorithms/category-outlier/IQR.ts:1]: The file name should be `lower camel case`, the rule is from @BBSQQ

File: packages/ava/src/insight/types.ts [hunk 1]
```diff
@@ -56,27 +70,26 @@ export type PatternInfo =
   | ChangePointInfo
   | CorrelationInfo;
 
-/** visualization for insight */
-export interface VisualizationSchema {
+/** explanation and visualization for insight */
+export interface NarrativeVisualizationSchema {
```
Comment (pddpd) [inline packages/ava/src/insight/types.ts:74]: Should we emphasize `Narrative`? It looks like text visualization

File: packages/ava/src/insight/visualization/index.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export function getInsightVisualizationSchema(
     schemas.push({
       chartType: chartType as ChartType,
       chartSchema,
-      caption: narrative.caption?.getContent(),
+      // @ts-ignore TODO @yuxi modify ntv generator and put caption into insightSummaries
```
Comment (pddpd) [inline packages/ava/src/insight/visualization/index.ts:55]: `@BBSQQ`

File: packages/ava/src/insight/visualization/index.ts [hunk 3]
```diff
@@ -135,7 +135,7 @@ export function getHomogeneousInsightVisualizationSchema(
     schemas.push({
       chartType,
       chartSchema,
-      caption: '',
+      // @ts-ignore TODO @yuxi modify ntv generator and put caption into insightSummaries
```
Comment (pddpd) [inline packages/ava/src/insight/visualization/index.ts:138]: `@BBSQQ`

File: packages/ava/src/insight/algorithms/index.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
-export * from './IQR';
-export * from './mannkendall-test';
+export * from './category-outlier';
+export * from './trend-direction';
```
Comment (pddpd) [inline packages/ava/src/insight/algorithms/index.ts:2]: want to know the naming thinking

File: package.json [hunk 5]
```diff
@@ -5,7 +5,6 @@
     "packages/*"
   ],
   "scripts": {
```
[the remark this answers is not in the document]
Comment (chenluli) [reply package.json:7]: I thought we do not use lerna now..

File: packages/ava/src/insight/algorithms/category-outlier/IQR.ts
Comment (chenluli) [reply packages/ava/src/insight/algorithms/category-outlier/IQR.ts:1]: there are many files and variables do not use `lower camel case` . such as:
截屏2023-03-17 下午4 26 45


should we change them all or only use this rule for new files?

File: packages/ava/src/insight/visualization/strategy/index.ts
Comment (BBSQQ) [inline packages/ava/src/insight/visualization/strategy/index.ts:1]: why empty

File: packages/ava/package.json [hunk 6]
```diff
@@ -54,6 +54,7 @@
   },
   "dependencies": {
     "@antv/antv-spec": "^0.1.0-alpha.18",
+    "@antv/g2": "^5.0.0-beta.13",
```
Comment (BBSQQ) [inline packages/ava/package.json:57]: 哦？之前 ava 都没有引入过 g2 嘛？

File: packages/ava/src/insight/types.ts [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import { PATTERN_TYPES, HOMOGENEOUS_PATTERN_TYPES } from './constant';
 
+import type { G2Spec } from '@antv/g2';
 import type { NtvTypes } from '../ntv';
+import type { DataTypes } from '../data';
```
Comment (BBSQQ) [inline packages/ava/src/insight/types.ts:5]: ava 内部推荐直接引用类型还是引用 namespace 啊？ @pddpd

File: packages/ava/src/insight/types.ts [hunk 8]
```diff
@@ -15,16 +17,28 @@ export type ImpactMeasureMethod = 'SUM' | 'COUNT';
 
 export type Aggregator = (data: Datum[], measure: string) => number;
 
-export type Subspace = {
-  dimension: string;
-  value: string;
-}[];
+export type DataProperty =
+  | (DataTypes.NumberFieldInfo & { name: string; fieldType: FieldType })
```
Comment (BBSQQ) [inline packages/ava/src/insight/types.ts:21]: 基于 NumberFieldInfo 扩展是不是太严格了？我看里面属性都是必选的

File: packages/ava/src/insight/types.ts [hunk 9]
```diff
@@ -56,22 +70,21 @@ export type PatternInfo =
...
-  chartSchema: any; // TODO type
+  chartSchema: G2Spec;
   /**
-   * @description insight summaries display type, it dependent
-   * @default string[]
+   * @description pure text or text schema to describe insight
+   * @default string
    */
-  insightSummaries?: string[] | NtvTypes.PhraseSpec[][];
+  insightSummaries?: string[] | NtvTypes.NarrativeTextSpec[];
```
Comment (BBSQQ) [inline packages/ava/src/insight/types.ts:81]: 是不是也叫 xxSchema 好一些呢？比如 explainSchema

File: packages/ava/src/insight/types.ts [hunk 10]
```diff
@@ -15,16 +17,28 @@ export type ImpactMeasureMethod = 'SUM' | 'COUNT';
...
+  | (DataTypes.StringFieldInfo & { name: string; fieldType: FieldType });
 
 export type Measure = {
+  /** field id, currently, field id is the same as the field name */
   field: string;
   method: MeasureMethod;
 };
 
+export type Dimension = {
+  /** field id, currently, field id is the same as the field name */
```
Comment (BBSQQ) [inline packages/ava/src/insight/types.ts:32]: fieldid === fieldname 是因为 insight 只接受 data 不接受 datameta。未来是不是有可能接受 meta 啊

File: packages/ava/src/insight/types.ts [hunk 11]
```diff
@@ -114,8 +135,8 @@ export interface HomogeneousPatternInfo {
   significance: number;
   insightType: InsightType;
   childPatterns: PatternInfo[];
-  exc?: string[];
-  commSet: string[];
+  exceptions?: string[];
+  commonSet: string[];
```
Comment (BBSQQ) [inline packages/ava/src/insight/types.ts:139]: 增加注释说明

File: packages/ava/src/insight/visualization/index.ts (hunk 2 above)
Comment (BBSQQ) [reply packages/ava/src/insight/visualization/index.ts:55]: ok

File: packages/ava/src/insight/visualization/index.ts (hunk 3 above)
Comment (BBSQQ) [reply packages/ava/src/insight/visualization/index.ts:138]: ok

File: packages/ava/src/insight/types.ts (hunk 7 above)
Comment (chenluli) [reply packages/ava/src/insight/types.ts]: 改成了直接引用

File: packages/ava/src/insight/types.ts (hunk 9 above)
Comment (chenluli) [reply packages/ava/src/insight/types.ts]: narrativeSchema

File: packages/ava/src/insight/algorithms/index.ts (hunk 4 above)
Comment (chenluli) [reply packages/ava/src/insight/algorithms/index.ts]: > want to know the naming thinking

change to single file and use the lower camel case

**Review by BBSQQ — APPROVED** [review]
LGTM

**Review by pddpd — APPROVED** [review]
LGTM

**Pull request merged**